com.unity.render-pipelines.universal/Runtime/UniversalProfiling.cs

```diff
@@ -0,0 +1,253 @@
+using System;
+using System.Collections.Generic;
+
+namespace UnityEngine.Rendering.Universal
+{
+    // These are tracked by the perf framework
+    internal enum URPProfileId
+    {
+        // CPU
+        UniversalRenderTotal,
+        UpdateVolumeFramework,
+        RenderCameraStack,
+
+        // GPU
+        AdditionalLightsShadow,
+        ColorGradingLUT,
+        CopyColor,
+        CopyDepth,
+        DepthNormalPrepass,
+        DepthPrepass,
+
+        // DrawObjectsPass
+        DrawOpaqueObjects,
+        DrawTransparentObjects,
+
+        // RenderObjectsPass
+        //RenderObjects,
+
+        MainLightShadow,
+        ResolveShadows,
+        SSAO,
+
+        // PostProcessPass
+        StopNaNs,
+        SMAA,
+        GaussianDepthOfField,
+        BokehDepthOfField,
+        MotionBlur,
+        PaniniProjection,
+        UberPostProcess,
+        Bloom,
+
+        FinalBlit
+    }
+
+    // A collection of profiling samplers, cache, helpers and convenience functions for Universal profiling.
+    // Keeps most of the profiling related things organized in one place.
+    // Static global state, mostly read/sometimes write, non-thread safe (like most things in Universal).
+    //
+    // Example:
+    // // Cached CPU
+    // using var profScope = UniversalProfiling.GetCPUScope(URPProfileId.DrawOpaqueObjects);
+    // using var profScope = UniversalProfiling.GetCPUScope(MethodBase.GetCurrentMethod());
+    // using var profScope = UniversalProfiling.GetCPUScope("example name");
+    // // Static GPU
+    // using( UniversalProfiling.GetGPUCPUScope( cmd, UniversalProfiling.Pipeline.ExampleStaticSampler ) ) { ... }
+    internal static class UniversalProfiling
```

Comment:
Yes, I agree. I didn't do that because it impacts a way larger amount of code. I'm trying to keep this localized for now. This class is independent of everything. 

I was going to propose an additional PR to move some of these into the ProfilingScope.
Thinking about this now, I think the string one is the only that really makes sense.
Some of these are URP specific like the camera. 
We might need to add one for RenderPasses if reflection doesn't work in practice.
The reflection issue kills the methodBase one too.

And in the end I suppose most of these can be additive changes, so perhaps it's not as disruptive as I initially thought.

Reply:
Removed.